### PR TITLE
[PWN-6854] Feature/pwn 6854

### PR DESCRIPTION
**Pull request by sidorov-panda**
## Link jira to issue
https://p2pvalidator.atlassian.net/browse/PWN-6854

## Description of the changes
-

### CODE REVIEW

File: p2p_wallet/Scenes/Main/Sell/info/MoonpayInfoView.swift [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                        )
+                }
+
+                VStack(alignment: .leading, spacing: 16) {
+                    VStack(alignment: .leading, spacing: 8) {
+                        Text("You will be redirected to our payment provider")
+                            .apply(style: .text1)
+                            .foregroundColor(Color(Asset.Colors.night.color))
+
+                        Text("You will need to enter your IBAN and pass KYC")
```
Comment (sidorov-panda) [inline p2p_wallet/Scenes/Main/Sell/info/MoonpayInfoView.swift:54]: localisation

File: p2p_wallet/Scenes/Main/Sell/info/MoonpayInfoView.swift [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                            .apply(style: .text4)
+                            .foregroundColor(Color(Asset.Colors.mountain.color))
+                    }
+
+                    VStack(alignment: .leading, spacing: 4) {
+                        Text("Transfer SOL to our payment provider from Key App")
```
Comment (sidorov-panda) [inline p2p_wallet/Scenes/Main/Sell/info/MoonpayInfoView.swift:60]: localisation

File: p2p_wallet/Scenes/Main/Sell/info/MoonpayInfoView.swift [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                        .frame(width: 24, height: 20)
+                        .background(
+                            Circle()
+                                .stroke(Color(Asset.Colors.mountain.color), lineWidth: 1.5)
+                        )
+                }
+
+                VStack(alignment: .leading, spacing: 16) {
+                    VStack(alignment: .leading, spacing: 8) {
+                        Text("You will be redirected to our payment provider")
```
Comment (sidorov-panda) [inline p2p_wallet/Scenes/Main/Sell/info/MoonpayInfoView.swift:50]: localisation

File: p2p_wallet/Scenes/Main/Sell/info/MoonpayInfoView.swift [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                        Text("You will need to enter your IBAN and pass KYC")
+                            .apply(style: .text4)
+                            .foregroundColor(Color(Asset.Colors.mountain.color))
+                    }
+
+                    VStack(alignment: .leading, spacing: 4) {
+                        Text("Transfer SOL to our payment provider from Key App")
+                            .apply(style: .text1)
+                            .foregroundColor(Color(Asset.Colors.night.color))
+                        Text("Nobody has access to your funds, so you need to execute the transaction cash out")
```
Comment (sidorov-panda) [inline p2p_wallet/Scenes/Main/Sell/info/MoonpayInfoView.swift:63]: localisation

File: p2p_wallet/Scenes/Main/Sell/SellCoordinator.swift [hunk 5]
```diff
@@ -148,6 +150,31 @@ final class SellCoordinator: Coordinator<SellCoordinatorResult> {
                 .handleEvents(receiveCompletion: { [unowned self] _ in
                     self.viewModel.warmUp()
                 }).eraseToAnyPublisher()
+
+        case .moonpayInfo:
+            moonpayInfoViewController = UIHostingController(
```
Comment (lisemyon) [inline p2p_wallet/Scenes/Main/Sell/SellCoordinator.swift:155]: I would have extracted it in separate coordinator to weight off this file but whatever) it is not critical

File: p2p_wallet/Scenes/Main/Sell/info/MoonpayInfoView.swift [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                onPressed: { actionButtonPressed?(isChecked) }
+            )
+            .frame(height: TextButton.Size.large.height)
+            .padding(.bottom, 30)
+            .padding(.horizontal, 16)
+        }
+    }
+}
+
+struct CheckboxView: View {
```
Comment (lisemyon) [inline p2p_wallet/Scenes/Main/Sell/info/MoonpayInfoView.swift:107]: separate file for this may be?

**Review by lisemyon — APPROVED** [review]

File: p2p_wallet/Scenes/Main/Sell/SellCoordinator.swift (hunk 5 above)
Comment (sidorov-panda) [reply p2p_wallet/Scenes/Main/Sell/SellCoordinator.swift:155]: didn't want to make one more coordinator that doesn't do any logic, only shows screen

File: p2p_wallet/Scenes/Main/Sell/info/MoonpayInfoView.swift (hunk 6 above)
Comment (sidorov-panda) [reply p2p_wallet/Scenes/Main/Sell/info/MoonpayInfoView.swift:107]: yes, good point

**Pull request merged**